geomstats/geometry/stratified/graph_space.py

```diff
@@ -381,141 +571,474 @@ def dist(self, graph_a, graph_b):
+        if attr_name in ["s_min", "s_max", "n_points"]:
+            self._s = None
+
+        return object.__setattr__(self, attr_name, value)
+
+    def _discretize_s(self):
+        r"""Compute the domain distretization."""
+        return gs.linspace(self.s_min, self.s_max, num=self.n_points)
+
+    @property
+    def s(self):
+        r"""Save the domain distretization."""
+        if self._s is None:
+            self._s = self._discretize_s()
+
+        return self._s
+
+    def _get_gamma_s(self, geodesic):
+        r"""Evaluate the geodesic in s."""
+        return geodesic(self.s)
+
+    def _compute_dists(self, metric, geodesic, x):
+        gamma_s = self._get_gamma_s(geodesic)
+
+        n_points = self._get_n_points(x)
+        if n_points > 1:
+            gamma_s = gs.repeat(gamma_s, n_points, axis=0)
+            rep_x = gs.concatenate([x for _ in range(self.n_points)])
+        else:
+            rep_x = x
+
+        dists = gs.reshape(metric.dist(gamma_s, rep_x), (self.n_points, n_points))
+
+        min_dists_idx = gs.argmin(dists, axis=0)
+
+        return dists, min_dists_idx, n_points
+
+    def dist(self, metric, geodesic, graph_to_permute):
+        r"""Compute the distance between the geodesic and the point.
+
+        Parameters
+        ----------
+        geodesic : function.
+        graph_to_permute : array-like, shape=[..., n, n]
+            Graph to align.
+
+        Returns
+        -------
+        dist : array-like, shape=[...,n]
+            Distance between the graph_to_permute and the geodesic.
+        """
+        dists, min_dists_idx, n_points = self._compute_dists(
+            metric, geodesic, graph_to_permute
+        )
+
+        return gs.take(
+            gs.transpose(dists),
+            min_dists_idx + gs.arange(n_points) * self.n_points,
+        )
+
+    def align(self, metric, geodesic, graph_to_permute):
+        r"""Align the graph to the geodesic.
+
+        Parameters
+        ----------
+        geodesic : function.
+        graph_to_permute : array-like, shape=[..., n, n]
+            Graph to align.
+
+        Returns
+        -------
+        permuted_graph : array-like, shape=[...,n]
+            Permuted graph as to be aligned with respect to the geodesic.
+        """
+        _, min_dists_idx, n_points = self._compute_dists(
+            metric, geodesic, graph_to_permute
+        )
+
+        perm_indices = min_dists_idx * n_points + gs.arange(n_points)
+        if n_points == 1:
+            perm_indices = perm_indices[0]
+
+        self.perm_ = gs.take(metric.perm_, perm_indices, axis=0)
+
+        return self._permute(metric, graph_to_permute, self.perm_)
+
+
+class _GeodesicToPointAligner(_BasePointToGeodesicAligner):
+    def __init__(self, method="BFGS", *, save_opt_res=False):
+        super().__init__()
+
+        self.method = method
+        self.save_opt_res = save_opt_res
+
+        self.opt_results_ = None
+
+    def _objective(self, s, metric, x, geodesic):
```

Comment:
what is s? what is x?

I know that this is a private class with a private method. yet, if future contributors are to collaborate on this code, having either docstrings or better variable names ( s or x does not give indication that helps parsing the code) could help understand the code.